From 8ff01a440f94660aba62550324a84212962ec254 Mon Sep 17 00:00:00 2001
From: Sergey Linev <S.Linev@gsi.de>
Date: Mon, 10 Jun 2024 09:19:47 +0200
Subject: [PATCH] [http] Allow to use any of openssl 3.x with civetweb

civetweb was designed when only openssl 3.0 was existing.
Therefore name of define is `OPENSSL_API_3_0`
Meanwhile there is openssl 3.1 and 3.2.
While seems to be there is no changes in API both can
be used with civetweb.
---
 net/http/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/http/CMakeLists.txt b/net/http/CMakeLists.txt
index 02d7514400c36..6ef98a604b5f6 100644
--- a/net/http/CMakeLists.txt
+++ b/net/http/CMakeLists.txt
@@ -78,7 +78,7 @@ if(ssl)
     MESSAGE(STATUS "Use SSL API VERSION 1.1 for civetweb")
     target_compile_definitions(RHTTP PUBLIC -DOPENSSL_API_1_1)
     set(link_ssl ON)
-  elseif((${ssl_major} EQUAL "3") AND ((${ssl_minor} EQUAL "0") OR (${ssl_minor} EQUAL "1")))
+  elseif(${ssl_major} EQUAL "3")
     MESSAGE(STATUS "Use SSL API VERSION 3.${ssl_minor} for civetweb")
     target_compile_definitions(RHTTP PUBLIC -DOPENSSL_API_3_0)
     set(link_ssl ON)
